### PR TITLE
Migrate to GitHub Actions Release Process

### LINKED ISSUE #79: Switch to GitHub Actions and goreleaser Release Process
The "standard library" Terraform Providers should implement nascent provider development tooling to encourage consistency, foster automation where possible, and discover bugs/enhancements to that tooling. To that end, this provider's release process should be switched to [`goreleaser`](http://goreleaser.com/) to match the [documented Terraform Registry publishing recommendations](https://www.terraform.io/docs/registry/providers/publishing.html). This includes:

- Creating necessary `.goreleaser.yml` and `.github/workflows/release.yml` configurations for tag-based releases (see also: TF-279 RFC)
- Ensuring necessary GitHub or Vault tokens are in place to fetch release secrets
- Ensuring provider and internal release process documentation is updated

### PR DESCRIPTION
**Pull request by bflad**
Closes #79

### CODE REVIEW

File: .goreleaser.yml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    - none
+signs:
+  - args: ["sign", "--dearmor", "--file", "${artifact}", "--out", "${signature}"]
+    artifacts: checksum
+    cmd: signore
+    signature: ${artifact}.sig
+  - args: ["sign", "--dearmor", "--file", "${artifact}", "--out", "${signature}"]
+    artifacts: checksum
+    cmd: signore
+    signature: ${artifact}.72D7468F.sig
```
Comment (anGie44) [inline .goreleaser.yml:66]: ahh so here you'll want to give one of the two sign configurations a unique ID else goreleaser (locally i'm using `v1.2.5`)  seems to fail with
```
$ goreleaser check
...
⨯ config is invalid error=found 2 signs with the ID 'default', please fix your config
   ⨯ command failed            error=invalid config: found 2 signs with the ID 'default', please fix your config
```

I've assumed the "fix" is to define an `id` key/value pair but if there's another way, lmk 😃 !

**Review by anGie44 — CHANGES_REQUESTED** [review]
One open comment, otherwise LGTM 👍

File: .goreleaser.yml (hunk 1 above)
Comment (bflad) [reply .goreleaser.yml:66]: Thank you for catching this! I do believe adding `id` to the second configuration there should resolve this. Will update locally and test, then push up if that error is gone. 👍

Comment (bflad) [reply .goreleaser.yml]: Updated!

File: .goreleaser.yml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+      # Ensure only built binary is archived
+      - 'none*'
+    format: zip
+    name_template: '{{ .ProjectName }}_{{ .Version }}_{{ .Os }}_{{ .Arch }}'
+before:
+  hooks:
+    - 'go mod download'
+builds:
+  - # Binary naming only required for Terraform CLI 0.12
+    binary: '{{ .ProjectName }}_v{{ .Version }}_x5'
```
Comment (anGie44) [inline .goreleaser.yml:12]: 👀 **TIL**

**Review by anGie44 — APPROVED** [review]

**Pull request merged**